### DISCUSSION TITLE
ScriptReference/AssetDatabase.CopyAsset

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#146] [Notes]
# ScriptReference/AssetDatabase.CopyAsset

https://docs.unity3d.com/ScriptReference/AssetDatabase.CopyAsset.html

**Comment by vertxxyz** [comment]
If you are using `CopyAsset`, and then [`DeleteAsset`](https://docs.unity3d.com/ScriptReference/AssetDatabase.DeleteAsset.html) for script files you need to call [`ImportAsset`](https://docs.unity3d.com/ScriptReference/AssetDatabase.ImportAsset.html) before deleting.  
If you do not call `ImportAsset` the `DeleteAsset` call will remove the asset, but the .csproj + cache will still think the script exists, and will complain: `error CS2001: Source file 'TScript.cs' could not be found`, forcing you to restart Unity to resolve the issue.
